### PR TITLE
Improve the GitHub Actions

### PR DESCRIPTION
**Pull request by PSchmiedmayer**
# *Name of the PR*

## :recycle: Current situation

1. The amazonlinux TestWebService tests seemed to have failed once. From the build logs it looked like the linux instance the action was running on was taking longer then expected to start up the web service.
2. Some runs of the CI did not use the cache to speed up the build process
3. The Codecov action did not report the test results reliably 

## :bulb: Proposed solution
This PRs address the issues noted above:
1. This PR increased the test timeout
2. This PR improves the caching mechanism
3. This PR addresses some error logs found in the Codecov action

### CODE REVIEW

File: .github/workflows/test-webservice.yml [hunk 1]
```diff
@@ -28,8 +29,8 @@ jobs:
         xcode-version: latest
     - uses: actions/cache@v2
       with:
-        path: .build
-        key: ${{ runner.os }}-spm-cachev2-${{ hashFiles('Package.resolved') }}
+        path: TestWebService/.build
```
Comment (bauer-andreas) [inline .github/workflows/test-webservice.yml:32]: Is the cache action not impacted by the `working-directory` definition?

**Review by bauer-andreas — APPROVED** [review]
looks good 👍 

Out of curiosity, which unit are those deadlines? minutes or seconds?

File: .github/workflows/test-webservice.yml (hunk 1 above)
Comment (PSchmiedmayer) [reply .github/workflows/test-webservice.yml:32]: Yes, it seems like it. The previous version didn't work and the cache only worked once a used absolute paths undefended of the workspace path.

Regarding the timeout: It is defined seconds, I think about 5 seconds should be reasonable to start up the web service 🚀

Comment (bauer-andreas) [reply .github/workflows/test-webservice.yml:32]: Weird behavior 😅 

Would it make sense to increase that even further? Should boot in 5 seconds, but we also have seen GitHub Actions being really slow 🙃 But lets just see how it goes.

Comment (PSchmiedmayer) [reply .github/workflows/test-webservice.yml:32]: Yes, I could also imaging extending the timeout even more if this is needed. On the other hand this gives us some hint about changes that dramatically increase the startup time of the web service so I would suggest to leave it at the current timeout for now and increase it once we have the need for it or we notice that builds start failing due to a timeout.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -15,17 +15,20 @@ jobs:
     name: macOS ${{ matrix.configuration }} test build
     runs-on: macos-latest
     strategy:
+      fail-fast: false
       matrix:
         configuration: [debug, release]
     steps:
     - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (tgymnich) [inline .github/workflows/build-and-test.yml:24]: @PSchmiedmayer why did you add this line? Is there a reason we need to fetch all history for all branches and tags?

Comment (PSchmiedmayer) [reply .github/workflows/build-and-test.yml:24]: @tgymnich There was a log error in the job that uploaded the codecov report that the action was requiring a fetch-depth of 0 or bigger then 1 and we had the issue that the report was not always reported in the PRs and after removing the log error the codecov reports continued to work again.

Comment (tgymnich) [reply .github/workflows/build-and-test.yml:24]: @PSchmiedmayer thanks for clarifying this!
